src/pages/search/Search.tsx

```diff
@@ -0,0 +1,169 @@
+import React, { useEffect, useState } from 'react';
+import {
+  IonContent,
+  IonHeader,
+  IonPage,
+  IonToolbar,
+  IonListHeader,
+  IonSearchbar,
+  IonList,
+  IonItem,
+  IonLabel,
+  IonThumbnail,
+  IonSegment,
+  IonTitle,
+  IonSegmentButton
+} from '@ionic/react';
+import data from "../../assets/data/data.json";
+import constitution from "../../assets/data/constitution.json";
+import './Search.css';
+import parse from 'html-react-parser';
+
+const Search: React.FC = () => {
+  const [isSearching, setIsSearching] = useState(false)
+  const [searchTerm, setSearchTerm] = useState("")
+  const [searchableProvisions, setSearchableProvisions] = useState<any>([]);
+  const [currentSegement, setCurrentSegment] = useState("all");
+
+  useEffect(() => {
+    const searchData = setupConstitutionSearch();
```

Comment:
As I understand it, this will be called every time the component mounts or is updated. It's expensive to setup the search data and it should only be done once. So either it needs to be setup outside of the FC (which is simplest right now), or this FC needs to be changed to a full Component and the setup done in the constructor.

Reply:
Hi Greg,  we pass [] as a dependency list in useEffect so that our setup only runs the first time the component is mounted.

Reply:
subtle!